python/labbox_ephys/api/_workersession.py

```diff
@@ -99,18 +99,31 @@ def handle_message(self, msg):
         elif type0 == 'hitherCreateJob':
             functionName = msg['functionName']
             kwargs = msg['kwargs']
-            opts = msg['opts']
             client_job_id = msg['clientJobId']
-            if opts.get('newHitherJobMethod', False):
-                try:
-                    job = hi.run(functionName, **kwargs, labbox=self._labbox_context).wait()
-                except Exception as err:
-                    self._send_message({
-                        'type': 'hitherJobCreationError',
-                        'client_job_id': client_job_id,
-                        'error': str(err) + ' (new method)'
-                    })
-                    return
+            try:
+                outer_job = hi.run(functionName, **kwargs, labbox=self._labbox_context)
+            except Exception as err:
+                self._send_message({
+                    'type': 'hitherJobError',
+                    'job_id': client_job_id,
+                    'client_job_id': client_job_id,
+                    'error_message': f'Error creating outer job: {str(err)}',
+                    'runtime_info': None
+                })
+                return
+            try:
+                job_or_result = outer_job.wait()
```

Comment:
We wait on the outer job to run. It will either return a hither job (to be queued) or an immediate result. If an immediate result, then we send back saying the job has finished. Otherwise, we will queue that job and send the finished message later.

Reply:
exactly.